std/src/main/scala/cats/std/map.scala

```diff
@@ -0,0 +1,32 @@
+package cats
+package std
+
+
+trait MapInstances {
+
+  implicit def MapInstance[K]: Traverse[Map[K, ?]] with FlatMap[Map[K, ?]] =
+    new Traverse[Map[K, ?]] with FlatMap[Map[K, ?]] {
+      override def traverse[G[_] : Applicative, A, B](fa: Map[K, A])(f: (A) => G[B]): G[Map[K, B]] = ???
+
+      override def flatMap[A, B](fa: Map[K, A])(f: (A) => Map[K, B]): Map[K, B] =
+        fa.flatMap { case (_, a) => f(a) }
+
+      override def foldLeft[A, B](fa: Map[K, A], b: B)(f: (B, A) => B): B =
+        fa.foldLeft(b) { case (x, (k, a)) => f(x, a)}
+
+      override def foldRight[A, B](fa: Map[K, A], b: B)(f: (A, B) => B): B =
+        fa.foldRight(b) { case ((k, a), z) => f(a, z)}
+
+      override def foldRight[A, B](fa: Map[K, A], b: Lazy[B])(f: (A, Lazy[B]) => B): Lazy[B] = {
+        // we use Lazy.byName(...) to avoid memoizing intermediate values.
+        def loop(as: Map[K, A], b: Lazy[B]): Lazy[B] =
+          as.toList match {
+            case Nil => b
+            case a :: rest => Lazy.byName(f(a._2, foldRight(rest.toMap, b)(f)))
```

Comment:
like @gbougeard I prefer `head :: tail` but your reason @non is persuasive and I think consistency is important and it is the reason why I choose `a :: rest`  ( because I hesitate to switch to `head :: tail` )